File: org.kevoree.genetic.framework/src/main/java/org/kevoree/genetic/framework/internal/RandomCompoundVariation.java
package org.kevoree.genetic.framework.internal;

import org.moeaframework.core.FrameworkException;
import org.moeaframework.core.Solution;
import org.moeaframework.core.Variation;
import org.moeaframework.core.operator.CompoundVariation;

import java.util.Arrays;
import java.util.Random;

/**
 * Created with IntelliJ IDEA.
 * User: duke
 * Date: 14/02/13
 * Time: 10:27
 */
public class RandomCompoundVariation extends CompoundVariation {

    Random rand = new Random();

    @Override
    public Solution[] evolve(Solution[] parents) {
        int indice = rand.nextInt(operators.size());
        return operators.get(indice).evolve(parents);
    }


}

File: org.kevoree.modeling.optimization.engine.genetic/src/main/java/org/kevoree/modeling/optimization/engine/genetic/impl/RandomCompoundVariation.kt
package org.kevoree.modeling.optimization.engine.genetic.impl

import org.moeaframework.core.operator.CompoundVariation
import org.moeaframework.core.Solution
import java.util.Random

/**
 * Created with IntelliJ IDEA.
 * User: duke
 * Date: 04/09/13
 * Time: 17:54
 */
class RandomCompoundVariation : CompoundVariation() {

    var rand = Random();

    public override fun evolve(parents: Array<out Solution>?): Array<Solution>? {
         var indice = rand.nextInt(operators!!.size());
        return operators!!.get(indice).evolve(parents);
    }
}